fetchai/ledger/api/tx.py

```diff
@@ -1,6 +1,33 @@
+import base64
+import binascii
+
 from .common import ApiEndpoint
 
 
+class TxStatus:
+    def __init__(self,
+                 digest: bytes,
+                 status: str,
+                 exit_code: int,
+                 charge: int,
+                 charge_rate: int,
+                 fee: int):
+        self._digest = digest
+        self.status = status
+        self.exit_code = exit_code
+        self.charge = charge
+        self.charge_rate = charge_rate
+        self.fee = fee
+
+    @property
+    def finished(self):
+        return self.status not in ('Unknown', 'Pending')
```

Comment:
This change has been changed around. I wonder if it is worth keeping the existing check, because I can only think of two exit state if you like for the transaction, where as we might want to have multiple in progress states for example

Reply:
I changed it and added the runtime error we talked about.